### PR TITLE
Warn about Proxies with version numbers different from the Broker

### PR DESCRIPTION
**Pull request by lablans**

### CODE REVIEW

File: shared/src/middleware.rs [hunk 1]
```diff
@@ -111,3 +111,26 @@ fn get_ip(req: &Request<Body>, info: &SocketAddr) -> IpAddr {
...
+pub async fn check_client_version(
+    req: Request<Body>,
+    next: Next<Body>,
+) -> Response {
+    let headers = req.headers();
+
+    if let Some(their_version_header) = headers.get(header::USER_AGENT) {
+        match compare_version(their_version_header) {
+            BeamWithMatchingVersion | NotBeam => {
+                // we're happy
```
Comment (Threated) [inline shared/src/middleware.rs:124]: Are we happy if its not beam or do we want to reject the request?

**Review by Threated — APPROVED** [review]

File: shared/src/middleware.rs (hunk 1 above)
Comment (Threated) [reply shared/src/middleware.rs]: I see

**Comment by lablans** [comment]
> It seems that it will only warn about the diffrent versions on the `v1/health` endpoint as only on thoese requests the proxy seems to add the special header.

@Threated , could you make sure the Proxy always sends its User-Agent header?

**Comment by TKussel** [comment]
@Threated This could tie in with the proposed/discussed Reqwest::Client change

**Comment by lablans** [comment]
@Threated Please check if this is still compatible to #154

**Pull request merged**